public/client.js

```diff
@@ -1,9 +1,46 @@
-// $(function () {
-// 	$.get('/trucks', function (trucks) {
-// 		var truckList = [];
-// 		trucks.forEach(function (truck) {
-// 			truckList.push('<li><a href="/trucks/' + truck.name + '/">' + truck.name + '</a></li>')
-// 		});
-// 		$('.truck-list').append(truckList);
-// 	});
-// });
\ No newline at end of file
+$(function () {
+	// Create a function for the truck list so that it can be called
+	// in order to switch between trucks and types.
+	// Immediately invoked to make truck list the default view
+	var truckView = (function truckView() {
```

Comment:
It looks like you meant to make truckView a function, but what's happening here is truckView is being assigned the result of calling your immediate executed function expression which doesn't explicitly return anything, therefore it returns undefined. If you remove the wrapping IIFE, you'll be able to toggle back and forth between the two views. The issue removing it introduces is that the truckView code isn't automatically called when the script runs, so you'll want to call it once explicitly.